### PR TITLE
Send trailers only when `TE: trailers` is set

### LINKED ISSUE #1709: Allow client to disable sending of trailers by omitting "TE" header.
## 🐛 Bug Report

We shouldn't send trailers unless the client sends the "TE" trailer.

See https://github.com/grpc-ecosystem/grpc-gateway/issues/1697 for more context

### PR DESCRIPTION
**Pull request by zoido**
#### References to other Issues or PRs
Fixes #1709

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?
Yes.

#### Brief description of what is fixed or changed

Sends trailers back only when client specifies `TE: trailers`. 

#### Other comments

I borrowed the code from `errors.go` from https://github.com/grpc-ecosystem/grpc-gateway/pull/1499 as well as the approach to the tests.
I was thinking about increasing the reuse but did not figure out anything elegant. Will gladly adjust.

### CODE REVIEW

File: examples/internal/integration/integration_test.go [hunk 1]
```diff
@@ -278,17 +280,27 @@ func testEchoOneof2(t *testing.T, port int, apiPrefix string, contentType string
...
+	req, err := http.NewRequest("POST", apiURL, bytes.NewReader(payload))
 	if err != nil {
-		t.Errorf("http.Post(%q) failed with %v; want success", apiURL, err)
+		t.Errorf("http.NewRequest() failed with %v; want success", err)
+		return
+	}
+	if useTrailers {
+		req.Header.Set("TE", "trailers")
+	}
+	resp, err := client.Do(req)
```
Comment (adambabik) [inline examples/internal/integration/integration_test.go:301]: You can skip declaration of `client`:

```suggestion
	resp, err := http.DefaultClient.Do(req)
```

**Review by adambabik — APPROVED** [review]
Thanks for the PR! LGTM

File: examples/internal/integration/integration_test.go (hunk 1 above)
Comment (zoido) [reply examples/internal/integration/integration_test.go]: Done. Thanks. Changed in the `testABELookupNotFound` as well.

File: runtime/handler.go [hunk 2]
```diff
@@ -137,6 +138,20 @@ func ForwardResponseMessage(ctx context.Context, mux *ServeMux, marshaler Marsha
...
 	handleForwardResponseServerMetadata(w, mux, md)
+
+	// RFC 7230 https://tools.ietf.org/html/rfc7230#section-4.1.2
+	// Unless the request includes a TE header field indicating "trailers"
+	// is acceptable, as described in Section 4.3, a server SHOULD NOT
+	// generate trailer fields that it believes are necessary for the user
+	// agent to receive.
+	var wantsTrailers bool
+
+	if te := req.Header.Get("TE"); strings.Contains(strings.ToLower(te), "trailers") {
```
[the remark this answers is not in the document]
Comment (zoido) [reply runtime/handler.go]: Done.

**Review by zoido — COMMENTED** [review]
Comments and suggestions addressed. PTAL.

**Comment by zoido** [comment]
Apologies for bothering. Are there any plans for this PR to be merged? Is there anything I should improve in order to enable the merge?

**Comment by johanbrandhorst** [comment]
> Apologies for bothering. Are there any plans for this PR to be merged? Is there anything I should improve in order to enable the merge?

Sorry this was delayed, thanks for your contribution!

**Pull request merged**